web_widget_image_webcam/static/src/js/webcam_widget.js

```diff
@@ -0,0 +1,106 @@
+                            Setting "force_flash" to "true" will always run in Adobe Flash fallback mode on Chrome, but it is not desirable.
+                        */
+                        force_flash: true,
+                        /*
+                            If "force_flash" is true, then this extra parameter "swfURL" is necessary.
+                        */
+                        swfURL: '/web_widget_image_webcam/static/src/js/webcam.swf',
+                    });
+                }
+            });
+
+            self.$el.find('.oe_form_binary_file_web_cam').click(function(){
+                // Init Webcam
+                Webcam.attach('#live_webcam');
+
+                // At time of Init "Save & Close" button is disabled
+                self.$el.find('#webcam_save_close').attr('disabled', 'disabled');
+
+                // Placeholder Image in the div "webcam_result"
+                self.$el.find("#webcam_result").html('<img src="/web_widget_image_webcam/static/src/img/webcam_placeholder.png"/>');
+            });
+
+            self.$el.find('#take_snap_btn').click(function(){
+                Webcam.snap( function(data) {
+                    img_data = data;
+                    // Display Snap besides Live WebCam Preview
+                    self.$el.find("#webcam_result").html('<img src="'+img_data+'"/>');
+                });
+                // Remove "disabled" attr from "Save & Close" button
+                self.$el.find('#webcam_save_close').removeAttr('disabled');
+            });
+
+            self.$el.find('#webcam_save_close').click(function(){
+                // Split `base64` string
+                var img_data_base64 = img_data.split(',')[1];
+
+                /*
+                    Size in base64 is approx 33% overhead the original data size.
+
+                    Source: -> http://stackoverflow.com/questions/11402329/base64-encoded-image-size
+                            -> http://stackoverflow.com/questions/6793575/estimating-the-size-of-binary-data-encoded-as-a-b64-string-in-python
+
+                            -> https://en.wikipedia.org/wiki/Base64
+                            [ The ratio of output bytes to input bytes is 4:3 (33% overhead).
+                            Specifically, given an input of n bytes, the output will be "4[n/3]" bytes long in base64,
+                            including padding characters. ]
+                */
+
+                // From the above info, we doing the opposite stuff to find the approx size of Image in bytes.
+                var approx_img_size = 3 * (img_data_base64.length / 4)  // like... "3[n/4]"
+
+                // Upload image in Binary Field
+                self.on_file_uploaded(approx_img_size, "web-cam-preview.jpeg", "image/jpeg", img_data_base64);
+
+                // Shut Down the Live Camera Preview | Reset the System
+                Webcam.reset();
+            });
+
+            self.$el.find('.webcam_close').click(function(){
```

Comment:
What if you press ESC to close the modal? Better bind on [`hide.bs.modal`](http://getbootstrap.com/javascript/#modals-events).